integration-tests/src/test_loop/utils/sharding.rs

```diff
@@ -118,3 +118,25 @@ pub fn shard_was_split(shard_layout: &ShardLayout, shard_id: ShardId) -> bool {
     };
     parent != shard_id
 }
+
+pub fn get_tracked_shards_from_prev_block(
+    client: &Client,
+    prev_block_hash: &CryptoHash,
+) -> Vec<ShardUId> {
+    let account_id =
+        client.validator_signer.get().map(|validator| validator.validator_id().clone());
+    let mut tracked_shards = vec![];
+    for shard_uid in
+        client.epoch_manager.get_shard_layout_from_prev_block(prev_block_hash).unwrap().shard_uids()
+    {
+        if client.shard_tracker.care_about_shard(
```

Comment:
There is a bit of a mismatch between the function name and the implementation here. 

The `care_about_shard` only checks if the client is a producer for given shard in this epoch. The client might also track a shard because it will be the producer in the next epoch or because it is configured to track all shards.

Reply:
`care_about_shard` also checks tracked shards, if `me` is `true`:
```
 match self.tracked_config {
            TrackedConfig::AllShards => {
                // Avoid looking up EpochId as a performance optimization.
                true
            }
            _ => self.tracks_shard(shard_id, parent_hash).unwrap_or(false),
        }
```